addons/web_tip/static/src/js/tip.js

```diff
@@ -2,57 +2,63 @@ odoo.define('web_tip.web_tip', function (require) {
 "use strict";
 
 var core = require('web.core');
-var form_widgets = require('web.form_widgets');
 var Model = require('web.DataModel');
-var utils = require('web.utils');
 var WebClient = require('web.WebClient');
+var formView = require('web.FormView');
+var utils = require('web.utils');
 
 var bus = core.bus;
 var Class = core.Class;
 
-var Tip = Class.extend({
+var Tips = Class.extend({
     init: function() {
         var self = this;
-        self.tips = [];
-        self.tip_mutex = new utils.Mutex();
-        self.$overlay = null;
-        self.$element = null;
+        var view_def = $.Deferred();
+
+        this.tips = [];
+        this.view = null;
 
         var Tips = new Model('web.tip');
-        Tips.query(['title', 'description', 'action_id', 'model', 'type', 'mode', 'trigger_selector',
+        Tips.query(['title', 'description', 'action_id', 'model', 'mode', 'trigger_selector',
             'highlight_selector', 'end_selector', 'end_event', 'placement', 'is_consumed'])
             .all().then(function(tips) {
                 self.tips = tips;
-            })
-        ;
+            });
 
-        core.bus.on('action', this, function(action) {
+        bus.on('action', this, function(action) {
             self.on_action(action);
         });
 
         bus.on('view_shown', this, function(view) {
             if (_.keys(view.fields_view).length === 0) {
                 view.on('view_loaded', this, function(fields_view) {
-                    self.on_view(view);
+                    self.view = view;
+                    view_def.resolve();
                 });
             } else {
-                self.on_view(view);
+                self.view = view;
+                view_def.resolve();
```

Comment:
What is the point of this deferred ? Why wouldn't it work with the old system of calling a function with view as argument ?